### PR TITLE
Fix T0010 FP: Allow delegate type parameter name

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Fixes
```cs
public delegate void SomeDelegate(int someParameterName);

public void Use(SomeDelegate action) { }

public void Test()
{
    Use(x => { }); // Compliant
    Use(someParameterName => { }); // FP
}
```

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.CSharp.Styling.Test/TestCases/LambdaParameterName.cs [hunk 1]
```diff
@@ -72,25 +73,52 @@ public void Initialize()
...
+        ParameterNamedTest delegate6 = testSome => { }; // Noncompliant
+
+        ParameterNamedCamelCasing delegate7 = camelCasing => { }; // Compliant
+        ParameterNamedCamelCasing delegate8 = camelcasing => { }; // Noncompliant
+        ParameterNamedCamelCasing delegate9 = camel => { };     // Noncompliant
+        ParameterNamedCamelCasing delegate10 = casing => { };    // Noncompliant
+
+        Func<int, int> function = arg => 0;  // Noncompliant, the delegate parameter is named "arg" (https://learn.microsoft.com/en-us/dotnet/api/system.func-2) but we do not allow that for Func<T, TResult>
+        Action<int> action = obj => { };     // Noncompliant, the delegate parameter is named "obj" (https://learn.microsoft.com/en-us/dotnet/api/system.action-1) but we do not allow that for Action<T>
+        new List<int>().Exists(obj => true); // Compliant. List.Exists uses System.Predicate<T> instead of System.Func<T, bool>
```
Comment (sebastien-marichal) [inline analyzers/tests/SonarAnalyzer.CSharp.Styling.Test/TestCases/LambdaParameterName.cs:122]: I think is an FN.

I don't know how often delegates are used in Collection type (and other types).
It looks to me, from the point of view of the coding style, we should use `x`.

I guess, it is an FN only when matching the delegate parameter name, so maybe it is fine.

**Review by sebastien-marichal — CHANGES_REQUESTED** [review]
I have one concern, and I would like to get your opinion.

File: analyzers/tests/SonarAnalyzer.CSharp.Styling.Test/TestCases/LambdaParameterName.cs (hunk 1 above)
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.CSharp.Styling.Test/TestCases/LambdaParameterName.cs:123]: Yes. This is very niche. `System.Predicate` is used by a couple of framework methods only. It predates `Action` and `Func` and is considered a legacy today. We could add it to `IsFuncOrAction`, but I do not think it is worth it. I just wanted to document it.

**Review by sebastien-marichal — APPROVED** [review]
LGTM!

**Pull request merged**